Sources/SwiftDocC/Model/Rendering/RenderContentCompiler.swift

```diff
@@ -47,7 +47,41 @@ struct RenderContentCompiler: MarkupVisitor {
     
     mutating func visitCodeBlock(_ codeBlock: CodeBlock) -> [any RenderContent] {
         // Default to the bundle's code listing syntax if one is not explicitly declared in the code block.
-        return [RenderBlockContent.codeListing(.init(syntax: codeBlock.language ?? bundle.info.defaultCodeListingLanguage, code: codeBlock.code.splitByNewlines, metadata: nil))]
+
+        if FeatureFlags.current.isExperimentalCodeBlockEnabled {
+
+            struct ParsedOptions {
+                var lang: String?
+                var nocopy = false
+            }
+
+            func parseLanguageString(_ input: String?) -> ParsedOptions {
```

Comment:
Do we have any plans to raise diagnostics here if we encounter an unknown option?

For example, the code below will parse `swift, nocpoy` as `lang: nocpoy`, `nocopy: false` and the only way for the developer to notice this is to see that the rendered code listing has a copy-to-clipboard button and doesn't have syntax highlighting. 

If we want the possibility of raising diagnostics from this (or future) code block options, it probably needs be parsed earlier in the build, before "rendering".

Reply:
I think it would be a good idea to raise diagnostics for unknown options. I’m not familiar with DocC diagnostics and I don’t understand why it would need to be emited earlier. Could you give me some more info?

Reply:
Thank you for all of this. Your explanation was very helpful.

I updated the PR with a new checker and tests for it, as well as adding an enum for the code block options directly on `CodeListing`. It should be ready for re-review.